driver/src/test/java/org/neo4j/driver/integration/TransactionIT.java

```diff
@@ -324,15 +320,13 @@ void shouldRollBackTxIfErrorWithConsume()
     }
 
     @Test
-    void shouldPropagateFailureFromSummary()
+    void shouldFailRun()
     {
         try ( Transaction tx = session.beginTransaction() )
         {
-            Result result = tx.run( "RETURN Wrong" );
+            ClientException e = assertThrows( ClientException.class, () -> tx.run( "RETURN Wrong" ) );
```

Comment:
that test would be a nice example to document the changed behavior for example.

Reply:
Yes, this demonstrates the main goal of this update well.